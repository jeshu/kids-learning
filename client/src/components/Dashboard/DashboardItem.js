import React from 'react'
import { Card, CardActionArea, CardActions, Button, CardContent, Typography, CardMedia } from "@material-ui/core"


function DashboardItem({path, title, desc, onPreview, onLaunch, id}) {
    return (title ? 
        <Card data-testid="dashboard-item">
            <CardActionArea>
                <CardMedia
                    image={path}
                    title={title}
                />
                <CardContent>
                    <Typography gutterBottom variant="h5">{title}</Typography>
                    <Typography variant="body2" color="textSecondary" component="p">{desc}</Typography>
                </CardContent>
            </CardActionArea>
            <CardActions>
                <Button title="preview" onClick={()=>onPreview({id})}>
                    Preview
                </Button>
                <Button title="launch" onClick={()=>{onLaunch({id})}}>
                    Launch
                </Button>
            </CardActions>
        </Card> : "");
}

export default DashboardItem
