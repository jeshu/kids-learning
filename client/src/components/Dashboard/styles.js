import { makeStyles } from "@material-ui/core";

export const useStyles = makeStyles({
    media: {
        maxWidth:"300px",
    },

    card: {
        width:"100px"
    }
})