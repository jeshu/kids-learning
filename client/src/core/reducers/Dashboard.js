const initialState = {

}

export default (state = initialState, { type, payload }) => {
    switch (type) {

        case "LOADING":
            return { ...state, loading: true }
            
        case "DASHBOARD_LOADED":
            return { ...state, ...payload, loading: false }
        default:
            return state
    }
}
