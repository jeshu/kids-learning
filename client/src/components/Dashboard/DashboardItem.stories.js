import React from 'react';
import DashboardItem from './DashboardItem';

export default { 
    title: "DashboardItem",
    component : DashboardItem
};

const props = {path:"", title:"Alphabets", desc:"Some desc", onPreview:()=>{}, onLaunch:()=>{}, id:"newID"}
export const AlphbetDashboardItem = () => <DashboardItem  {...props}/>
AlphbetDashboardItem.story = {
    name : "Alphabets",
}