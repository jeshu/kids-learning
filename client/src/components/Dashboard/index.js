import React from 'react';
import DashboardItem from './DashboardItem';
import { Grid, Container } from '@material-ui/core';

function Dashboard(props) {
    return props ? (
        <React.Fragment>
            <Container>
                <Grid container spacing={3}>
                    {props.data && props.data.map(item =>
                        <Grid key={item.id} item xs={12} md={6} lg={3} >
                            <DashboardItem  data-testid="dashboard-item" {...item} />
                        </Grid>
                    )}
                </Grid>
            </Container>
        </React.Fragment>
    ) : "";
}

export default Dashboard
