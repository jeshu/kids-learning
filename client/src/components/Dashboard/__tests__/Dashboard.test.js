import React from 'react';
import Dashboard from "..";
import { render } from "@testing-library/react";


describe('test suit for dashboard', () => {
    const props = {
        data:[{
            title:"title-1",
            path:"path-1",
            desc:"desc-1",
            id:"id-1"
        },{
            title:"title-2",
            path:"path-2",
            desc:"desc-2",
            id:"id-2"
        },{
            title:"title-3",
            path:"path-3",
            desc:"desc-3",
            id:"id-3"
        }]
    }
    it('should render without errors but not buttons', () => {
        const { container } = render(<Dashboard />)
        expect(container).toMatchSnapshot();
    });

    it('should render the DashboardItems in dom', ()=>{
        const { getAllByTestId } = render(<Dashboard {...props} />);
        expect(getAllByTestId("dashboard-item").length).toBe(3);
    })

});
