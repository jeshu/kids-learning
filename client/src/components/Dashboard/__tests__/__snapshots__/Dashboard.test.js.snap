// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test suit for dashboard should render without errors but not buttons 1`] = `
<div>
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3"
    />
  </div>
</div>
`;
